### DISCUSSION TITLE
when returning big bytes using StreamResponse, time cost goes unacceptable

### DISCUSSION THREAD

**Discussion by wuutiing** [fastapi/fastapi#8465] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.responses import StreamingResponse
from io import BytesIO

app = FastAPI()


@app.get('/test1')
async def test1():
  mimetype = 'image/png'
  imgio = BytesIO(open('<abspath of a big image>', 'rb').read())
  return StreamingResponse(imgio, media_type=mimetype)


def _yield_bytes(path):
  with open(path, 'rb') as f:
    yield from f


@app.get('/test2')
async def test2():
  mimetype = 'image/png'

  return StreamingResponse(_yield_bytes('<abspath of a big image>'),
                           media_type=mimetype)


'''
gunicorn -b 0.0.0.0:8001 \
  -k uvicorn.workers.UvicornWorker \
  -w 1 \
  test_fastapi:app
'''
```

### Description

I'm rewriting a flak project which returns bytes using `send_file` in flask style code
and a fastapi impl is like the code above, which then tested slow(a 12MB image costs more than 20s).
test time snippet is simple
```
import time
start = time.time()
requests.get('http://localhost:8001/test1')
end = time.time()
print(end-start) # more than 20s for a 6k image, about 12MB
```

also the flask style code is provided here (a same image costs less than 0.5s)
```
from flask import Flask, send_file
from io import BytesIO

app = Flask('test_flask')


@app.route('/test')
def test():
  mimetype = 'image/png'
  imgio = BytesIO(open('<abspath of a big image>','rb').read())
  return send_file(imgio, mimetype=mimetype, cache_timeout=0)


'''
gunicorn -b 0.0.0.0:8000 \
  -w 1 \
  test_flask:app
'''
```

### Operating System

macOS

### Operating System Details

    Hardware Overview:

      Model Name: MacBook Pro
      Model Identifier: MacBookPro14,1
      Processor Name: Dual-Core Intel Core i5
      Processor Speed: 2.3 GHz
      Number of Processors: 1
      Total Number of Cores: 2
      L2 Cache (per Core): 256 KB
      L3 Cache: 4 MB
      Hyper-Threading Technology: Enabled
      Memory: 8 GB
      System Firmware Version: 447.40.12.0.0
      OS Loader Version: 540.40.4~45
      SMC Version (system): 2.43f11
      Serial Number (system): C02VLPR1HV29

### FastAPI Version

0.75.2

### Python Version

3.7.6

**Comment by jgould22** [comment]
Why are you using streaming response to send a file?

How does it work if you use file response ?
https://fastapi.tiangolo.com/advanced/custom-response/#fileresponse

**Comment by jgould22** [comment, marked as answer]
Ok I checked it and yes FileResponse is much faster.

This stack overflow post will explain to you why 

https://stackoverflow.com/questions/55873174/how-do-i-return-an-image-in-fastapi

**Comment by wuutiing** [comment]
> Ok I checked it and yes FileResponse is much faster.
> 
> This stack overflow post will explain to you why 
> 
> https://stackoverflow.com/questions/55873174/how-do-i-return-an-image-in-fastapi

thank you, the refer helps much. the reason why not use FileResponse is that  the file
-open is used just for convenience, data is all in memory not disk.
thank you again
